src/diagnostics.rs

```diff
@@ -118,12 +118,16 @@ pub fn report_error<'tcx, 'mir>(
     report_msg(ecx, &format!("{}: {}", title, msg), msg, helps, true);
 
     // Extra output to help debug specific issues.
-    if let UndefinedBehavior(UndefinedBehaviorInfo::InvalidUndefBytes(Some(ptr))) = e.kind {
+    if let UndefinedBehavior(UndefinedBehaviorInfo::InvalidUndefBytes {
+            ptr: Some(p),
+            size: Some(s),
+        }) = e.kind {
```

Comment:
At this point, a `match` could be more readable, as it separates pattern and scrutinee better?